framework/src/outputs/XDA.C

```diff
@@ -44,15 +44,17 @@ XDA::XDA(const InputParameters & parameters) :
 void
 XDA::output(const ExecFlagType & /*type*/)
 {
+  std::string fname = filename();
+  std::string base = fname.substr(0, fname.size()-4);
```

Comment:
The second argument here could be negative. Since len is of type `size_t` this could be bad. I suggest a mooseError or some other sanity check just in case.